src/pybind/mgr/dashboard/controllers/orchestrator.py

```diff
@@ -122,10 +122,13 @@ def identify_device(self, hostname, device, duration):  # pragma: no cover
 class OrchestratorInventory(RESTController):
 
     @raise_if_no_orchestrator([OrchFeature.DEVICE_LIST])
-    def list(self, hostname=None):
+    def list(self, hostname=None, refresh=None):
         orch = OrchClient.instance()
         hosts = [hostname] if hostname else None
-        inventory_hosts = [host.to_json() for host in orch.inventory.list(hosts)]
+        do_refresh = False
+        if refresh is not None:
```

Comment:
Wouldn't it make sense to enhance `str_to_bool` to return `False` if `refresh=None`. This will reduce the amount of code here.

```
def str_to_bool(val):
	if isinstance(val, bool):
		return val
 	if val is None:
		return False  # bool(None) => False
	return bool(strtobool(val))
```

Reply:
Hmm. I'm not quite sure about this. There are many uses of this function in the backend codes.
One usage is to convert the string to boolean after checking the input parameter (like `refresh` here).
Maybe it's better to let the caller determine what to do when the parameter is not defined.
The caller might want to do the refresh when the `refresh` parameter is not provided.